unofficial/c511027106.lua

```diff
@@ -21,31 +21,29 @@ function s.initial_effect(c)
 	e2:SetProperty(EFFECT_FLAG_IGNORE_IMMUNE)
 	e2:SetValue(s.val)
 	c:RegisterEffect(e2)
-    --destroy replace
-    local e3=Effect.CreateEffect(c)
-    e3:SetType(EFFECT_TYPE_FIELD+EFFECT_TYPE_CONTINUOUS)
-    e3:SetCode(EFFECT_DESTROY_REPLACE)
-    e3:SetRange(LOCATION_SZONE)
-    e3:SetTarget(s.reptg)
-    e3:SetValue(s.repval)
-    e3:SetOperation(s.repop)
-    c:RegisterEffect(e3)
+    	--destroy replace
+    	local e3=Effect.CreateEffect(c)
+    	e3:SetType(EFFECT_TYPE_FIELD+EFFECT_TYPE_CONTINUOUS)
+    	e3:SetCode(EFFECT_DESTROY_REPLACE)
+    	e3:SetRange(LOCATION_SZONE)
+    	e3:SetTarget(s.reptg)
+    	e3:SetValue(s.repval)
+    	e3:SetOperation(s.repop)
+    	c:RegisterEffect(e3)
 end
 
 s.listed_names={56840427}
 
 --equip
 
-function s.filter(c)
-	return c:IsFaceup() and c:IsCode(56840427)
-end
-
 function s.eqtg(e,tp,eg,ep,ev,re,r,rp,chk,chkc)
-	if chkc then return chkc:IsLocation(LOCATION_MZONE) and chkc:IsControler(tp) and s.filter(chkc) end
+	if chkc then return chkc:IsLocation(LOCATION_MZONE) and chkc:IsControler(tp) end
 	if chk==0 then return Duel.GetLocationCount(tp,LOCATION_SZONE)>0
 		and Duel.IsExistingTarget(s.filter,tp,LOCATION_MZONE,0,1,nil) end
 	Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_EQUIP)
-	Duel.SelectTarget(tp,s.filter,tp,LOCATION_MZONE,0,1,1,nil)
+	Duel.SelectTarget(tp,aux.FilterFaceupFunction(Card.IsCode,56840427),tp,LOCATION_MZONE,0,1,1,nil)
+	Duel.SetOperationInfo(0,CATEGORY_EQUIP,Duel.GetFirstTarget(),1,0,0)
+	

```

Comment:
Should be "e:GetHandler()", not "Duel.GetFirstTarget" (the card that is being equipped to another card, not the card that is being equipped with this card). Also, there's an extra empty line here.